lib/Doctrine/ORM/Proxy/Factory/StaticProxyFactory.php

```diff
@@ -186,6 +207,44 @@ private function transientFieldsFqns(ClassMetadata $metadata) : array
         return $transientFieldsFqns;
     }
 
+    private function collectionFieldsFqns(ClassMetadata $metadata) : array
+    {
+        $transientFieldsFqns = [];
+
+        foreach ($metadata->getDeclaredPropertiesIterator() as $name => $property) {
+            if (! $property instanceof ToManyAssociationMetadata) {
+                continue;
+            }
+
+            $transientFieldsFqns[] = $this->propertyFqcn(
+                $property
+                    ->getDeclaringClass()
+                    ->getReflectionClass()
+                    ->getProperty($name) // @TODO possible NPR. This should never be null, why is it allowed to be?
```

Comment:
and it's not. unless you got a bug. =\

Reply:
Yes, the problem is that `getProperty()` can lead to a NPE here. Would probably need an API that gives me a reflection property or throws

Reply:
Indeed, accessor that throws and has a non-null return type is needed.

Reply:
Get method should throw imho, unlike find.